Makefile

```
@@ -234,6 +234,11 @@ archi: $(EXTRA_DIR)/ocamldot/ocamldot
 	echo "}" >> archi.dot
 	dot -Tpdf archi.dot > archi.pdf
 
+lock:
+	opam lock ./alt-ergo-lib.opam -w
+	# Remove OCaml compiler constraints
+	sed -i '/\"ocaml/d' ./alt-ergo-lib.opam.locked
```

We shouldn't be blanket removing packages that start with `"ocaml` because there are non-compiler-related libraries whose name starts with `ocaml` (e.g. `ocamlgraph`).

In our case, it removes the lock for `ocaml-compiler-libs`, `ocamlbuild` and `ocamlfind` which I believe are independent of the compiler version.

I think the following is better:
```suggestion
	sed -i '/"ocaml"\|"ocaml-base-compiler"\|"ocaml-system"\|"ocaml-config"/d' ./alt-ergo-lib.opam.locked
```